packages/ses/src/commons.js

```diff
@@ -360,3 +360,25 @@ export const FERAL_STACK_GETTER = feralStackGetter;
  * @type {((newValue: any) => void) | undefined}
  */
 export const FERAL_STACK_SETTER = feralStackSetter;
+
+// Test for async generator function syntax support.
+let AsyncGeneratorNewFunctionInstance;
+try {
+  // Wrapping one in an new Function lets the `hermesc` binary file
+  // parse the Metro js bundle without SyntaxError, to generate the
+  // optimised Hermes bytecode bundle, when `gradlew` is called to
+  // assemble the release build APK for React Native prod Android apps.
+  // Delaying the error until runtime lets us customise lockdown behaviour.
+  AsyncGeneratorNewFunctionInstance = new FERAL_FUNCTION(
+    'return (async function* AsyncGeneratorFunctionInstance() {})',
+  )();
+} catch (e) {
+  if (e.name === 'SyntaxError') {
```

Comment:
nb: if stricter is better

```suggestion
  if (Error.prototype.jsEngine === 'hermes' && e.name === 'SyntaxError' && e.message.includes('async generators are unsupported')) {
```

Reply:
Maybe going half-way is a decent compromise (thinking matching on freeform text in error messages might be brittle)?

```suggestion
  if (Error.prototype.jsEngine === 'hermes' && e.name === 'SyntaxError') {
```

Reply:
altho this PR covers Hermes compiler with Hermes VM disabled, i retested latest changes with Hermes VM enabled, had to git bisect to find `Error.prototype.jsEngine === 'hermes'` causing the following `print` output to fail and swallowed `async generators are unsupported` error to be re-thrown :suspect:

my mistake, `Error.prototype.jsEngine` is only set by React Native runtime, _not_ Hermes

```cpp
// https://github.com/facebook/react-native/blob/main/packages/react-native/ReactCommon/hermes/executor/HermesExecutorFactory.cpp#L224-L230

// ...

  // Add js engine information to Error.prototype so in error reporting we
  // can send this information.
  auto errorPrototype =
      decoratedRuntime->global()
          .getPropertyAsObject(*decoratedRuntime, "Error")
          .getPropertyAsObject(*decoratedRuntime, "prototype");
  errorPrototype.setProperty(*decoratedRuntime, "jsEngine", "hermes");


// ...
```

removed in https://github.com/endojs/endo/pull/2334/commits/f543b169e040572de545a1ce6c0900e4e37920f8 and documented in https://github.com/endojs/endo/pull/2334/commits/2dc851c0bfa1971f89cc561b0f7b4279cccfcd96